src/client/pkg/grpcutil/stream.go

```diff
@@ -123,3 +123,44 @@ func WriteFromStreamingBytesClient(streamingBytesClient StreamingBytesClient, wr
 	}
 	return nil
 }
+
+// copyBuffer is the same as io.CopyBuffer except that it always uses the
+// given buffer.  In contract, io.CopyBuffer does not use the given buffer
+// if the reader has WriteTo defined.
+func copyBuffer(dst io.Writer, src io.Reader, buf []byte) (written int64, err error) {
+	if buf != nil && len(buf) == 0 {
+		panic("empty buffer in io.CopyBuffer")
+	}
+
+	if buf == nil {
+		buf = make([]byte, 32*1024)
+	}
+
+	for {
+		nr, er := src.Read(buf)
+		if nr > 0 {
+			nw, ew := dst.Write(buf[0:nr])
+			if nw > 0 {
+				written += int64(nw)
+			}
+
+			if ew != nil {
+				err = ew
+				break
+			}
+
+			if nr != nw {
+				err = io.ErrShortWrite
+				break
+			}
+		}
+
+		if er != nil {
+			if er != io.EOF {
+				err = er
+			}
+			break
+		}
+	}
+	return written, err
+}
```

Comment:
It seems like an easier way to implement this function would be to just wrap the `io.Reader` in type that doesn't implement the `WriteTo` method but implements `Read` as a pass through to `src.Read` then you wouldn't need any extra code.